src/cdk/testing/testbed/fake-events/type-in-element.ts

```diff
@@ -22,6 +22,50 @@ const incrementalInputTypes = new Set([
   'url',
 ]);
 
+/**
+ * Manual mapping of some common characters to their `code` in a keyboard event. Non-exhaustive, see
+ * the tables on MDN for more info: https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/keyCode
+ */
+const charsToCodes: Record<string, string> = {
+  ' ': 'Space',
+  '.': 'Period',
+  ',': 'Comma',
+  '`': 'Backquote',
+  '-': 'Minus',
+  '=': 'Equal',
+  '[': 'BracketLeft',
+  ']': 'BracketRight',
+  '\\': 'Backslash',
+  '/': 'Slash',
+  "'": 'Quote',
+  '"': 'Quote',
+  ';': 'Semicolon',
+};
+
+/**
+ * Determines the `KeyboardEvent.key` from a character. See #27034 and
+ * https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/code
+ */
+function getKeyboardEventCode(char: string): string {
+  if (char.length !== 1) {
+    return '';
+  }
+
+  const charCode = char.charCodeAt(0);
+
+  // Key is a letter between a and z, uppercase or lowercase.
+  if ((charCode >= 97 && charCode <= 122) || (charCode >= 65 && charCode <= 90)) {
+    return `Key${char.toUpperCase()}`;
+  }
+
+  // Digits from 0 to 9.
+  if (48 <= charCode && charCode <= 57) {
+    return `Digit${char}`;
+  }
+
+  return charsToCodes[char] ?? '';
```

Comment:
Random thought: we could return `<unsupported>` here instead which would make it more obvious that our code doesn't have a mapping for the specific character.

Reply:
Sounds like a good idea to me, could potentially prevent some bad tests